src/test/scala/kamon/opentracing/test/TestReporter.scala

```diff
@@ -0,0 +1,53 @@
+package kamon.opentracing
+package test
+
+import com.typesafe.config.Config
+import kamon.trace.Span.{FinishedSpan => KamonFinishedSpan}
+import kamon.{SpanReporter => KamonSpanReporter}
+
+import scala.concurrent.duration.{Duration, DurationInt}
+
+object TestReporter extends KamonSpanReporter {
+  case class AnnotationReport(name: String, fields: Map[String, String])
+  case class Report(operationName: String, traceId: String, spanId: String, parentId: String, tags: Map[String, Any], annotations: Seq[AnnotationReport])
+
+  private object Lock
+  private val DefaultDuration = 2 seconds
+  private var reportStack: List[KamonFinishedSpan] = List.empty
+
+  override def start(): Unit = ()
+  override def stop(): Unit = ()
+  override def reconfigure(config: Config): Unit = ()
+  override def reportSpans(spans: Seq[KamonFinishedSpan]): Unit = Lock.synchronized {
+    reportStack = spans ++: reportStack
+  }
+
+  def getLatestReport: Option[Report] = Lock.synchronized {
+    reportStack match {
+      case head :: tail =>
+        reportStack = tail
+        Some(fromFinishedSpan(head))
+      case Nil => None
+    }
+  }
+
+  def waitForReport(timeout: Duration = DefaultDuration): Option[Report] = {
```

Comment:
I think this one is unnecessary, just use the one above that returns Option[Report] and use `eventually` blocks in the test.